lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -318,7 +318,11 @@ def ensure(module, base, state, names, exclude):
     allow_erasing = False
 
     if exclude:
-        base.conf.exclude_pkgs(exclude)
+        try:
+            base.conf.exclude_pkgs(exclude)
+        except AttributeError:
+            module.fail_json(msg="Could not exclude packages."
+                                 " Please update `dnf` package.")
```

Comment:
Are user's likely to upgrade the dnf package to 2.1.x from whatever they have installed?  It sounds like a major upgrade.... If so, it might be better to phrase this as "The `exclude` option requires dnf version 2.1.0 or higher" rather than asking them to upgrade.